File: 68/solution.py
from typing import List


class Solution:
    def fullJustify(self, words: List[str], maxWidth: int) -> List[str]:
        res = []
        i = 0
        n = len(words)
        while i < n:
            left = i
            l = 0
            while i < n and l + len(words[i]) + i - left <= maxWidth:
                l += len(words[i])
                i += 1
            black_count = maxWidth - l
            if i == n:
                res.append(" ".join(words[left:]) + " "*(black_count - i + left + 1))
                break
            count = i - left
            if count == 1:
                res.append(words[left] + " "*black_count)
            else:
                item = ""
                avg = black_count // (count - 1)
                more = black_count % (count - 1)
                for j in range(more):
                    item += words[left + j] + " " * (avg + 1)
                res.append(item + (" "*avg).join(words[left + more:i]))
        return res

File: 68/solution.go
package _68

import "strings"

func fullJustify(words []string, maxWidth int) []string {
	res := []string{}
	i := 0
	n := len(words)
	for i < n {
		left := i
		l := 0
		for i < n && l+len(words[i])+i-left <= maxWidth {
			l += len(words[i])
			i++
		}
		black_count := maxWidth - l
		if i == n {
			res = append(res, strings.Join(words[left:], " ")+strings.Repeat(" ", black_count-i+left+1))
			break
		}
		count := i - left
		if count == 1 {
			res = append(res, words[left]+strings.Repeat(" ", black_count))
		} else {
			item := ""
			avg := black_count / (count - 1)
			more := black_count % (count - 1)
			for j := 0; j < more; j++ {
				item += words[left+j] + strings.Repeat(" ", avg+1)
			}
			res = append(res, item+strings.Join(words[left+more:i], strings.Repeat(" ", avg)))
		}
	}
	return res
}
